enhancements/per-node-configuration.md

```diff
@@ -0,0 +1,601 @@
+spec:
+  nodeSelector:
+    matchLabels:
+      security-zone: dmz
+  hostsession:
+    asn: 64512
+    hostasn: 64710
+---
+# L3Passthrough for internal nodes
+apiVersion: openpe.openperouter.github.io/v1alpha1
+kind: L3Passthrough
+metadata:
+  name: internal-passthrough
+  namespace: openperouter-system
+spec:
+  nodeSelector:
+    matchLabels:
+      security-zone: internal
+  hostsession:
+    asn: 64512
+    hostasn: 64720
+```
+
+#### Multiple Instances on Same Node
+
+Example showing multiple L3VNIs configured on the same set of nodes for different tenants:
+
+```yaml
+# Tenant A VNI on worker nodes
+apiVersion: openpe.openperouter.github.io/v1alpha1
+kind: L3VNI
+metadata:
+  name: tenant-a-vni
+  namespace: openperouter-system
+spec:
+  nodeSelector:
+    matchLabels:
+      node-role.kubernetes.io/worker: ""
+  vrf: tenant-a
+  vni: 5001
+  hostsession:
+    asn: 64512
+    hostasn: 64600
+---
+# Tenant B VNI on the same worker nodes
+apiVersion: openpe.openperouter.github.io/v1alpha1
+kind: L3VNI
+metadata:
+  name: tenant-b-vni
+  namespace: openperouter-system
+spec:
+  nodeSelector:
+    matchLabels:
+      node-role.kubernetes.io/worker: ""
+  vrf: tenant-b
+  vni: 5002
+  hostsession:
+    asn: 64512
+    hostasn: 64601
+---
+# Tenant C VNI on the same worker nodes
+apiVersion: openpe.openperouter.github.io/v1alpha1
+kind: L3VNI
+metadata:
+  name: tenant-c-vni
+  namespace: openperouter-system
+spec:
+  nodeSelector:
+    matchLabels:
+      node-role.kubernetes.io/worker: ""
+  vrf: tenant-c
+  vni: 5003
+  hostsession:
+    asn: 64512
+    hostasn: 64602
+```
+
+In this example, all worker nodes (with `node-role.kubernetes.io/worker` label) will have three L3VNIs configured, one for each tenant. This is allowed and expected behavior for L3VNI, L2VNI, and L3Passthrough resources.
+
+### Controller Implementation
+
+#### Node Matching Logic
+
+All controllers (Underlay, L3VNI, L2VNI, and L3Passthrough) will be enhanced with:
+
+1. **Node Label Watching**: Add a watch for Node resource label changes to trigger reconciliation
+2. **Selector Matching**: For each node, determine which resource instance applies by matching node labels against all resource `nodeSelector` fields
+3. **Configuration Application**: Generate and apply configuration only on nodes matched by the resource's selector
+4. **Dynamic Updates**: Reconcile when:
+   - Resources are created/updated/deleted
+   - Node labels change
+   - Nodes are added/removed from the cluster
+
+#### Conflict Resolution
+
+Conflict resolution uses **shared validation logic** between the admission webhook and reconcile loop. Controllers protect themselves without depending on webhooks.
+
+**Underlay-specific restriction:**
+
+Only one Underlay can match a given node. Enforced at two levels using shared validation code:
+
+1. **Admission webhook**: First-line defense that will validate the current resource received at the webhook request without looking at other resurces.
```

Comment:
seems that we already are looking into other resources: https://github.com/openperouter/openperouter/blob/3ecbac7c2f4eb398b41505e76a58e955c07b7575/internal/webhooks/l2vni_webhook.go#L104

I please put this back the way it was. Sorry for only having noticed it now. 
I'll lgtm it afterwards.

Reply:
I am going to remove it, we need to not break what we have already, but that's implementation details.